sdk/src/main/java/com/supenta/flitchio/sdk/Status.java

```diff
@@ -0,0 +1,165 @@
+     * @see #failureReason
+     * @since 0.7.0
+     */
+    public static final int BINDING_FAILED = 2;
+
+    /**
+     * The {@link FlitchioController} has successfully bound to the Flitchio Manager app.
+     *
+     * @since 0.7.0
+     */
+    public static final int BOUND = 3;
+
+    /**
+     * The {@link FlitchioController} is bound to the Flitchio Manager app and Flitchio is
+     * connected to the phone. You can call {@link FlitchioController#obtainSnapshot()} to poll
+     * data. You will also start getting event callbacks if you have declared a
+     * {@link FlitchioEventListener}.
+     *
+     * @since 0.7.0
+     */
+    public static final int CONNECTED = 4;
+
+    /**
+     * The {@link FlitchioController} is bound to the Flitchio Manager app and Flitchio is
+     * disconnected from the phone.
+     *
+     * @since 0.7.0
+     */
+    public static final int DISCONNECTED = 5;
+
+    /**
+     * Identifier for this status.
+     * Its value is one of the {@link Status} constants.
+     *
+     * @since 0.7.0
+     */
+    public final int code;
+
+    /**
+     * If this status' {@link #code} is {@link #BINDING_FAILED}, this holds the reason of the
+     * failure and its value is either one of {@link FailingStatus#REASON_MANAGER_UNUSABLE},
+     * {@link FailingStatus#REASON_SERVICE_REFUSED_CONNECTION},
+     * {@link FailingStatus#REASON_SERVICE_UNREACHABLE} or
+     * {@link FailingStatus#REASON_SERVICE_SHUTDOWN_CONNECTION}.
+     * If this status' {@link #code} is not {@link #BINDING_FAILED}, its value is
+     * {@link FailingStatus#REASON_NONE}.
+     *
+     * @since 0.7.0
+     */
+    public final int failureReason;
+
+    /**
+     * Create a new non-failing status.
+     *
+     * @param code Identifier for this status: one of the {@link Status} constants.
+     */
+    Status(int code) {
+        this(code, FailingStatus.REASON_NONE);
+    }
+
+    private Status(int code, int failureReason) {
+        this.code = code;
+        this.failureReason = failureReason;
+
+        if (this.code == BINDING_FAILED && this.failureReason == FailingStatus.REASON_NONE) {
+            throw new IllegalArgumentException("Declared a BINDING_FAILED status without providing a failure reason");
+        } else if (this.code != BINDING_FAILED && this.failureReason != FailingStatus.REASON_NONE) {
+            throw new IllegalArgumentException("Declared status is not BINDING_FAILED but a failure reason has been provided");
+        }
+    }
+
+    /**
+     * Subclass of {@link Status} describing that a failure has happened.
+     * A failing status has a {@link #BINDING_FAILED} code and it holds the reason of the
+     * failure in {@link #failureReason}.
+     *
+     * @since 0.7.0
+     */
+    public static class FailingStatus extends Status {
+        /**
+         * Default failure reason when the status is not failing.
+         *
+         * @since 0.7.0
+         */
+        public static final int REASON_NONE = -1;
```

Comment:
Probably better to rename this to something like `NOT_FAILED` because now it's like saying that it has failed but for no reason. It will help differentiate it as the fallback I think.